packages/grafana-data/src/types/datasource.ts

```diff
@@ -306,6 +306,11 @@ abstract class DataSourceApi<
    */
   metricFindQuery?(query: any, options?: LegacyMetricFindQueryOptions): Promise<MetricFindValue[]>;
 
+  /**
+   * Verify adhoc filters applicability based on queries
```

Comment:
TODO - update this to :
`Verify adhoc filters applicability based on queries and filters`!